src/hashtable.c

```diff
@@ -79,7 +79,8 @@ static pair_t *hashtable_find_pair(hashtable_t *hashtable, bucket_t *bucket,
     list = bucket->first;
     while (1) {
         pair = list_to_pair(list);
-        if (pair->hash == hash && strcmp(pair->key, key) == 0)
+        if (pair->hash == hash && pair->key_len == key_len &&
+            strncmp(pair->key, key, key_len) == 0)
```

Comment:
this `strncmp()` should be `memcmp()` in case there is a null in the middle (and because we already know the lengths match)

Reply:
Replaced, thanks for your review!